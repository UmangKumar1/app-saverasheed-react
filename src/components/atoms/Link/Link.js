import PropTypes from 'prop-types'
import { Link as RRLink } from 'react-router-dom'
import styled from 'react-emotion'

const { string, any } = PropTypes

const Link = styled(RRLink)({
  textDecoration: 'none',
  color: 'inherit',
})

Link.propTypes = {
  children: any,
  to: string.isRequired,
}

export default Link
