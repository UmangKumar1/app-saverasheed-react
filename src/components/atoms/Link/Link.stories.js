import React from 'react'
import { storiesOf } from '@storybook/react'
import Link from '.'

storiesOf('Link', module).add('with text', () => (
  <Link to="foo">Save Rasheed</Link>
))
