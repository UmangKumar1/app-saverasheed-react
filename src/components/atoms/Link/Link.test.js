import React from 'react'
import { shallow } from 'enzyme'
import Link from '.'

const wrap = (props = {}) => shallow(<Link {...props} />)

describe('Link', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ children: 'Link', to: 'foo' })
    expect(wrapper).toMatchSnapshot()
  })
})
