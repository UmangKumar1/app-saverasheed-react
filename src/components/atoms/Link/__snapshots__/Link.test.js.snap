// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(Link)
    replace={false}
    to="foo"
  >
    Link
  </Styled(Link)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Link",
      "className": "css-nn640c",
      "replace": false,
      "to": "foo",
    },
    "ref": null,
    "rendered": "Link",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Link",
        "className": "css-nn640c",
        "replace": false,
        "to": "foo",
      },
      "ref": null,
      "rendered": "Link",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
